atom: 1.27.0 -> 1.27.1, atom-beta: 1.24.0-beta3 -> 1.28.0-beta1

Synthetica9 opened the pull request:
Also rewrote the derivation for easier maintencence.

(And added @ysndr as maintainer for atom-beta, kind of an implicit
result of joining the two. If you're not okay with that, I'll patch
you out, but I'll presume consent.)

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

GrahamcOfBorg commented:
No attempt on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40886&attempt_id=9168bd22-8061-4fa7-a301-d5024a44d77f)

The following builds were skipped because they don't evaluate on aarch64-linux: atom

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40886&attempt_id=24f5cfbc-0870-4094-85de-bbeda43939d2)

Attempted: atom

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/3flf6ll6i935d1v98wvr36yd9xf5q9wb-atom-1.27.1/share/atom/resources/app/apm/node_modules/node-gyp/gyp/gyp: interpreter directive changed from "/bin/sh" to "/nix/store/xn5gv3lpfy91yvfy9b0i7klfcxh9xskz-bash-4.4-p19/bin/sh"
/nix/store/3flf6ll6i935d1v98wvr36yd9xf5q9wb-atom-1.27.1/share/atom/resources/app/apm/node_modules/open/vendor/xdg-open: interpreter directive changed from "/bin/sh" to "/nix/store/xn5gv3lpfy91yvfy9b0i7klfcxh9xskz-bash-4.4-p19/bin/sh"
/nix/store/3flf6ll6i935d1v98wvr36yd9xf5q9wb-atom-1.27.1/share/atom/resources/app/apm/node_modules/ajv/scripts/prepare-tests: interpreter directive changed from "/usr/bin/env sh" to "/nix/store/xn5gv3lpfy91yvfy9b0i7klfcxh9xskz-bash-4.4-p19/bin/sh"
/nix/store/3flf6ll6i935d1v98wvr36yd9xf5q9wb-atom-1.27.1/share/atom/resources/app/apm/node_modules/ajv/scripts/travis-gh-pages: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/xn5gv3lpfy91yvfy9b0i7klfcxh9xskz-bash-4.4-p19/bin/bash"
/nix/store/3flf6ll6i935d1v98wvr36yd9xf5q9wb-atom-1.27.1/bin/.atom-wrapped: interpreter directive changed from "/bin/bash" to "/nix/store/xn5gv3lpfy91yvfy9b0i7klfcxh9xskz-bash-4.4-p19/bin/bash"
checking for references to /build in /nix/store/3flf6ll6i935d1v98wvr36yd9xf5q9wb-atom-1.27.1...
cannot find section .dynamic
file /nix/store/3flf6ll6i935d1v98wvr36yd9xf5q9wb-atom-1.27.1/share/atom/atom had a PT_GNU_STACK program header, converted
file /nix/store/3flf6ll6i935d1v98wvr36yd9xf5q9wb-atom-1.27.1/share/atom/resources/app/apm/bin/node had a PT_GNU_STACK program header, converted
/nix/store/3flf6ll6i935d1v98wvr36yd9xf5q9wb-atom-1.27.1
```
</p></details>

GrahamcOfBorg commented:
No attempt on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40886&attempt_id=ea5b87b7-916d-4b1a-9484-32161341cad0)

The following builds were skipped because they don't evaluate on aarch64-linux: atom

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40886&attempt_id=912002e8-488e-4e25-afdd-1cadf2ea79a3)

Attempted: atom

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/s9m9j0kjvq3jscx5mzjhn8d59h2g7q1v-atom-1.27.1/share/atom/resources/app/apm/node_modules/ajv/scripts/prepare-tests: interpreter directive changed from "/usr/bin/env sh" to "/nix/store/xn5gv3lpfy91yvfy9b0i7klfcxh9xskz-bash-4.4-p19/bin/sh"
/nix/store/s9m9j0kjvq3jscx5mzjhn8d59h2g7q1v-atom-1.27.1/share/atom/resources/app/apm/node_modules/open/vendor/xdg-open: interpreter directive changed from "/bin/sh" to "/nix/store/xn5gv3lpfy91yvfy9b0i7klfcxh9xskz-bash-4.4-p19/bin/sh"
/nix/store/s9m9j0kjvq3jscx5mzjhn8d59h2g7q1v-atom-1.27.1/share/atom/resources/app/apm/node_modules/keytar/script/cibuild: interpreter directive changed from "/bin/bash" to "/nix/store/xn5gv3lpfy91yvfy9b0i7klfcxh9xskz-bash-4.4-p19/bin/bash"
/nix/store/s9m9j0kjvq3jscx5mzjhn8d59h2g7q1v-atom-1.27.1/share/atom/resources/app/apm/script/postinstall.sh: interpreter directive changed from "/bin/bash" to "/nix/store/xn5gv3lpfy91yvfy9b0i7klfcxh9xskz-bash-4.4-p19/bin/bash"
/nix/store/s9m9j0kjvq3jscx5mzjhn8d59h2g7q1v-atom-1.27.1/share/atom/resources/app/atom.sh: interpreter directive changed from "/bin/bash" to "/nix/store/xn5gv3lpfy91yvfy9b0i7klfcxh9xskz-bash-4.4-p19/bin/bash"
checking for references to /build in /nix/store/s9m9j0kjvq3jscx5mzjhn8d59h2g7q1v-atom-1.27.1...
cannot find section .dynamic
file /nix/store/s9m9j0kjvq3jscx5mzjhn8d59h2g7q1v-atom-1.27.1/share/atom/atom had a PT_GNU_STACK program header, converted
file /nix/store/s9m9j0kjvq3jscx5mzjhn8d59h2g7q1v-atom-1.27.1/share/atom/resources/app/apm/bin/node had a PT_GNU_STACK program header, converted
/nix/store/s9m9j0kjvq3jscx5mzjhn8d59h2g7q1v-atom-1.27.1
```
</p></details>

xeji commented:
@GrahamcOfBorg build atom-beta

GrahamcOfBorg commented:
No attempt on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40886&attempt_id=2570a77f-0e51-4b38-91a9-4b9b6535029a)

The following builds were skipped because they don't evaluate on aarch64-linux: atom-beta

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
No attempt on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40886&attempt_id=7c5e9e5c-d236-40a0-b586-b7d3a791da2f)

The following builds were skipped because they don't evaluate on x86_64-darwin: atom-beta

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40886&attempt_id=fd7202d6-a676-4fa3-8261-c258f31ffd8b)

Attempted: atom-beta

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/0s6jlnwphbr77gs99aizfw6n8nxb2c7d-atom-beta-1.28.0-beta1/share/atom-beta/resources/app/apm/node_modules/ajv/scripts/travis-gh-pages: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/xn5gv3lpfy91yvfy9b0i7klfcxh9xskz-bash-4.4-p19/bin/bash"
/nix/store/0s6jlnwphbr77gs99aizfw6n8nxb2c7d-atom-beta-1.28.0-beta1/share/atom-beta/resources/app/apm/node_modules/ajv/scripts/prepare-tests: interpreter directive changed from "/usr/bin/env sh" to "/nix/store/xn5gv3lpfy91yvfy9b0i7klfcxh9xskz-bash-4.4-p19/bin/sh"
/nix/store/0s6jlnwphbr77gs99aizfw6n8nxb2c7d-atom-beta-1.28.0-beta1/share/atom-beta/resources/app/apm/node_modules/open/vendor/xdg-open: interpreter directive changed from "/bin/sh" to "/nix/store/xn5gv3lpfy91yvfy9b0i7klfcxh9xskz-bash-4.4-p19/bin/sh"
/nix/store/0s6jlnwphbr77gs99aizfw6n8nxb2c7d-atom-beta-1.28.0-beta1/share/atom-beta/resources/app/apm/script/postinstall.sh: interpreter directive changed from "/bin/bash" to "/nix/store/xn5gv3lpfy91yvfy9b0i7klfcxh9xskz-bash-4.4-p19/bin/bash"
/nix/store/0s6jlnwphbr77gs99aizfw6n8nxb2c7d-atom-beta-1.28.0-beta1/share/atom-beta/resources/app/atom.sh: interpreter directive changed from "/bin/bash" to "/nix/store/xn5gv3lpfy91yvfy9b0i7klfcxh9xskz-bash-4.4-p19/bin/bash"
checking for references to /build in /nix/store/0s6jlnwphbr77gs99aizfw6n8nxb2c7d-atom-beta-1.28.0-beta1...
cannot find section .dynamic
ln: failed to create symbolic link '/nix/store/0s6jlnwphbr77gs99aizfw6n8nxb2c7d-atom-beta-1.28.0-beta1/share/atom/resources/app.asar.unpacked/node_modules/dugite/git/bin/git': No such file or directory
builder for '/nix/store/wp8y4pv5g3hvl5ckmj6ngajmd6qdwwv5-atom-beta-1.28.0-beta1.drv' failed with exit code 1
error: build of '/nix/store/wp8y4pv5g3hvl5ckmj6ngajmd6qdwwv5-atom-beta-1.28.0-beta1.drv' failed
```
</p></details>

xeji reviewed (commented):
Please fix the build of atom-beta (it fails on my local machine too). 
The atom version is 1.27.1 but your PR and the commit messages say 1.27.7.

Synthetica9 committed:
atom, atom-beta: 1.27.0 -> 1.27.1, 1.24.0-beta3 -> 1.28.0-beta1

Also rewrote the derivation for easier maintencence.

(And added @ysndr as maintainer for atom-beta, kind of an implicit
result of joining the two. If you're not okay with that, I'll patch
you out, but I'll presume consent.)

GrahamcOfBorg commented:
No attempt on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40886&attempt_id=b254edc6-f11c-4abd-a2c8-a1d78a94fc4d)

The following builds were skipped because they don't evaluate on aarch64-linux: atom, atom-beta

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnsupportedSystem = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnsupportedSystem = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40886&attempt_id=c70ca7ed-7d72-40b0-bb18-597c5f8a3151)

Attempted: atom, atom-beta

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/6s4nhhgsrw97gr48673x80zywscxvqax-atom-beta-1.28.0-beta1/share/atom-beta/resources/app/apm/node_modules/open/vendor/xdg-open: interpreter directive changed from "/bin/sh" to "/nix/store/xn5gv3lpfy91yvfy9b0i7klfcxh9xskz-bash-4.4-p19/bin/sh"
/nix/store/6s4nhhgsrw97gr48673x80zywscxvqax-atom-beta-1.28.0-beta1/share/atom-beta/resources/app/apm/node_modules/ajv/scripts/prepare-tests: interpreter directive changed from "/usr/bin/env sh" to "/nix/store/xn5gv3lpfy91yvfy9b0i7klfcxh9xskz-bash-4.4-p19/bin/sh"
/nix/store/6s4nhhgsrw97gr48673x80zywscxvqax-atom-beta-1.28.0-beta1/share/atom-beta/resources/app/apm/node_modules/ajv/scripts/travis-gh-pages: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/xn5gv3lpfy91yvfy9b0i7klfcxh9xskz-bash-4.4-p19/bin/bash"
/nix/store/6s4nhhgsrw97gr48673x80zywscxvqax-atom-beta-1.28.0-beta1/bin/.atom-beta-wrapped: interpreter directive changed from "/bin/bash" to "/nix/store/xn5gv3lpfy91yvfy9b0i7klfcxh9xskz-bash-4.4-p19/bin/bash"
checking for references to /build in /nix/store/6s4nhhgsrw97gr48673x80zywscxvqax-atom-beta-1.28.0-beta1...
cannot find section .dynamic
file /nix/store/6s4nhhgsrw97gr48673x80zywscxvqax-atom-beta-1.28.0-beta1/share/atom-beta/atom had a PT_GNU_STACK program header, converted
file /nix/store/6s4nhhgsrw97gr48673x80zywscxvqax-atom-beta-1.28.0-beta1/share/atom-beta/resources/app/apm/bin/node had a PT_GNU_STACK program header, converted
/nix/store/ayvsqrca7zscxvcw2apl60l5zi4v38na-atom-1.27.1
/nix/store/6s4nhhgsrw97gr48673x80zywscxvqax-atom-beta-1.28.0-beta1
```
</p></details>

Synthetica9 commented:
@xeji :heavy_check_mark:
